### PR TITLE
[BugFix] Clear destination of CTE after retrying a query

### LINKED ISSUE #15183: [sqlancer] Unknown error of "java.lang.IllegalStateException: null"
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'
```sql
DROP DATABASE IF EXISTS v0;
CREATE DATABASE v0;
USE v0;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATE NOT NULL,c_0_1 LARGEINT NOT NULL,c_0_2 BOOLEAN NOT NULL,c_0_3 BIGINT NOT NULL,c_0_4 VARCHAR(11) NOT NULL,c_0_5 VARCHAR(21) NOT NULL,c_0_6 LARGEINT NOT NULL,c_0_7 LARGEINT NOT NULL,c_0_8 INT NOT NULL,c_0_9 VARCHAR(11) NOT NULL,c_0_10 BOOLEAN NOT NULL,c_0_11 DATE NOT NULL,c_0_12 DOUBLE NOT NULL,c_0_13 ARRAY<INT> NOT NULL,c_0_14 DATETIME NOT NULL, INDEX index1_c_0_4(`c_0_4`) USING BITMAP) PRIMARY KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7,c_0_8,c_0_9) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 103 day)) DISTRIBUTED BY HASH (c_0_7,c_0_6,c_0_1,c_0_9,c_0_8,c_0_4,c_0_5,c_0_2,c_0_3,c_0_0) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_0_4, c_0_9") ;
CREATE TABLE t1 (c_1_0 DATE NULL ,c_1_1 DATETIME NOT NULL ) DUPLICATE KEY (c_1_0) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 96 day)) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t0 VALUES ('2010-01-28', 1917342145, false, 479590212, 'Z', '', -1020795475, -1020795475, 24, '', false, '2010-01-19', 0.8965089594498077, [1061801370, 1917342145, -289753130, -289753130, -1914914851, -1, 2134797324, -1020795475], '2010-01-28 08:00:14'), ('2010-01-02', 89409176, false, -1317718529, 'Znb3P', 'ICtaz', -1350323103, -1317718529, 29, 'vP', true, '2010-01-28', -1.914914851E9, [1061801370, 1911757292, 60441437, -248538979, -660202134, -1483868731, -1786977635, -1506454067, -1020795475], '2010-01-04 08:00:14');
INSERT INTO t0 VALUES ('2010-01-08', -586353843, true, -1786977635, 'NqVP5XjN', 'RVuRm', -338826418, 1697395427, 29, '', false, '2010-01-07', 0.32508042352668143, [-338826418, -1350323103, 1512536381, 89409176, 1494257042, -162778857, 1697496042, 1381456038, 1668535628, 1911757292], '2010-01-17 08:00:14'), ('2009-12-31', 60441437, false, 1571878667, 'K6VzN', 'W9', 420138878, -723063118, 17, '8y', true, '2010-01-14', 0.8965089594498077, [443859998, -680970715, 479590212, -1639982907, 1594486499, -338826418, -162778857, 1697395427], '2010-01-05 08:00:14');
INSERT INTO t0 VALUES ('2010-01-26', -346595995, true, -1506454067, '', 'rmUhBOGJX', -1432604900, -607176421, 4, '', false, '2010-01-06', 0.36383887067247433, [1783329456, 1196620932, 2134797324], '2010-01-19 08:00:14'), ('2010-01-21', 1571878667, true, -1302374786, 'ag', '1da', -10194904, -949610295, 22, '2zhrI5', false, '2010-01-01', 0.5396362805247813, [-616383529, 1512536381, -616383529, 1061801370, 759024468, -1639982907, 1457540823, 582518229, 1697496042, 286358167, 2134797324, -1914914851, -368002688, -1221828821, 144606043], '2010-01-11 08:00:14');
TRUNCATE TABLE t1;
CREATE VIEW v0 AS SELECT t1_3.c_1_0 FROM t1 AS t1_3;
CREATE VIEW v1 AS SELECT t1_3.c_1_0 FROM t1 AS t1_3 WHERE true;
CREATE VIEW v2 AS SELECT t1_3.c_1_0 FROM t1 AS t1_3 WHERE (NOT (true));
CREATE VIEW v3 AS SELECT t1_3.c_1_0 FROM t1 AS t1_3 WHERE ((true) IS NULL);
CREATE VIEW v4 AS ((SELECT t1_3.c_1_0 FROM t1 AS t1_3 WHERE true) UNION ALL (SELECT t1_3.c_1_0 FROM t1 AS t1_3 WHERE (NOT (true))) UNION ALL (SELECT t1_3.c_1_0 FROM t1 AS t1_3 WHERE ((true) IS NULL)) );
CREATE VIEW v5 AS WITH with_t_0 as (SELECT v4_28.c_1_0 FROM v4 AS v4_28 WHERE false) SELECT v0_24.c_1_0 FROM v0 AS v0_24, t0 AS t0_22, v2 AS v2_26, v4 AS v4_28, t1 AS t1_23, v1 AS v1_25, with_t_0 AS with_t_0;
CREATE VIEW v6 AS WITH with_t_0 as (SELECT v4_28.c_1_0 FROM v4 AS v4_28 WHERE false) SELECT v0_24.c_1_0 FROM v0 AS v0_24, t0 AS t0_22, v2 AS v2_26, v4 AS v4_28, t1 AS t1_23, v1 AS v1_25, with_t_0 AS with_t_0 WHERE ( ( CAST(t0_22.c_0_4 AS INT )  ) = ( t0_22.c_0_7 ) ) IN (true, false) ;
CREATE VIEW v7 AS WITH with_t_0 as (SELECT v4_28.c_1_0 FROM v4 AS v4_28 WHERE false) SELECT v0_24.c_1_0 FROM v0 AS v0_24, t0 AS t0_22, v2 AS v2_26, v4 AS v4_28, t1 AS t1_23, v1 AS v1_25, with_t_0 AS with_t_0 WHERE (NOT (( ( CAST(t0_22.c_0_4 AS INT )  ) = ( t0_22.c_0_7 ) ) IN (true, false) ));
CREATE VIEW v8 AS WITH with_t_0 as (SELECT v4_28.c_1_0 FROM v4 AS v4_28 WHERE false) SELECT v0_24.c_1_0 FROM v0 AS v0_24, t0 AS t0_22, v2 AS v2_26, v4 AS v4_28, t1 AS t1_23, v1 AS v1_25, with_t_0 AS with_t_0 WHERE ((( ( CAST(t0_22.c_0_4 AS INT )  ) = ( t0_22.c_0_7 ) ) IN (true, false) ) IS NULL);
CREATE VIEW v9 AS (WITH with_t_0 as (SELECT v4_28.c_1_0 FROM v4 AS v4_28 WHERE false) (SELECT v0_24.c_1_0 FROM v0 AS v0_24, t0 AS t0_22, v2 AS v2_26, v4 AS v4_28, t1 AS t1_23, v1 AS v1_25, with_t_0 AS with_t_0 WHERE ( ( CAST(t0_22.c_0_4 AS INT )  ) = ( t0_22.c_0_7 ) ) IN (true, false) ) UNION ALL (SELECT v0_24.c_1_0 FROM v0 AS v0_24, t0 AS t0_22, v2 AS v2_26, v4 AS v4_28, t1 AS t1_23, v1 AS v1_25, with_t_0 AS with_t_0 WHERE (NOT (( ( CAST(t0_22.c_0_4 AS INT )  ) = ( t0_22.c_0_7 ) ) IN (true, false) ))) UNION ALL (SELECT v0_24.c_1_0 FROM v0 AS v0_24, t0 AS t0_22, v2 AS v2_26, v4 AS v4_28, t1 AS t1_23, v1 AS v1_25, with_t_0 AS with_t_0 WHERE ((( ( CAST(t0_22.c_0_4 AS INT )  ) = ( t0_22.c_0_7 ) ) IN (true, false) ) IS NULL)) );
CREATE VIEW v10 AS SELECT v3_72.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, v4 AS v4_73, t1 AS t1_68, v9 AS v9_78, v7 AS v7_76, v5 AS v5_74, v2 AS v2_71, v8 AS v8_77;
CREATE VIEW v11 AS SELECT v3_72.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, v4 AS v4_73, t1 AS t1_68, v9 AS v9_78, v7 AS v7_76, v5 AS v5_74, v2 AS v2_71, v8 AS v8_77 WHERE (v2_71.c_1_0) IN ('1969-12-23', '1970-01-07', '1969-12-10') ;
CREATE VIEW v12 AS SELECT v3_72.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, v4 AS v4_73, t1 AS t1_68, v9 AS v9_78, v7 AS v7_76, v5 AS v5_74, v2 AS v2_71, v8 AS v8_77 WHERE (NOT ((v2_71.c_1_0) IN ('1969-12-23', '1970-01-07', '1969-12-10') ));
CREATE VIEW v13 AS SELECT v3_72.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, v4 AS v4_73, t1 AS t1_68, v9 AS v9_78, v7 AS v7_76, v5 AS v5_74, v2 AS v2_71, v8 AS v8_77 WHERE (((v2_71.c_1_0) IN ('1969-12-23', '1970-01-07', '1969-12-10') ) IS NULL);
CREATE VIEW v14 AS ((SELECT v3_72.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, v4 AS v4_73, t1 AS t1_68, v9 AS v9_78, v7 AS v7_76, v5 AS v5_74, v2 AS v2_71, v8 AS v8_77 WHERE (v2_71.c_1_0) IN ('1969-12-23', '1970-01-07', '1969-12-10') ) UNION ALL (SELECT v3_72.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, v4 AS v4_73, t1 AS t1_68, v9 AS v9_78, v7 AS v7_76, v5 AS v5_74, v2 AS v2_71, v8 AS v8_77 WHERE (NOT ((v2_71.c_1_0) IN ('1969-12-23', '1970-01-07', '1969-12-10') ))) UNION ALL (SELECT v3_72.c_1_0 FROM v3 AS v3_72, t0 AS t0_67, v4 AS v4_73, t1 AS t1_68, v9 AS v9_78, v7 AS v7_76, v5 AS v5_74, v2 AS v2_71, v8 AS v8_77 WHERE (((v2_71.c_1_0) IN ('1969-12-23', '1970-01-07', '1969-12-10') ) IS NULL)) );
CREATE VIEW v15 AS SELECT subt1.c_1_0 FROM v8 AS v8_152, v12 AS v12_143,  (SELECT t1_138.c_1_0, t1_138.c_1_1 FROM t1 AS t1_138 WHERE ('1969-12-13') IN ((SELECT v14_145.c_1_0 FROM v14 AS v14_145, v5 AS v5_149, v3 AS v3_147, v1 AS v1_140 WHERE NULL) ) ) subt1 INNER JOIN (SELECT v0_139.c_1_0 FROM v0 AS v0_139) subv0 ON subt1.c_1_1 = subv0.c_1_0 AND subt1.c_1_1 < subv0.c_1_0,  v11 AS v11_142 INNER JOIN (SELECT v13_144.c_1_0 FROM v13 AS v13_144) subv13 ON v11_142.c_1_0 = subv13.c_1_0 AND '1969-12-13' <= subv13.c_1_0 AND v11_142.c_1_0 = subv13.c_1_0 AND v11_142.c_1_0 > subv13.c_1_0,  t0 AS t0_137 RIGHT ANTI JOIN v4 AS v4_148 ON t0_137.c_0_3 = v4_148.c_1_0 AND t0_137.c_0_3 != v4_148.c_1_0 AND t0_137.c_0_6 > v4_148.c_1_0 AND t0_137.c_0_6 < v4_148.c_1_0;
CREATE VIEW v16 AS SELECT subt1.c_1_0 FROM v8 AS v8_152, v12 AS v12_143,  (SELECT t1_138.c_1_0, t1_138.c_1_1 FROM t1 AS t1_138 WHERE ('1969-12-13') IN ((SELECT v14_145.c_1_0 FROM v14 AS v14_145, v5 AS v5_149, v3 AS v3_147, v1 AS v1_140 WHERE NULL) ) ) subt1 INNER JOIN (SELECT v0_139.c_1_0 FROM v0 AS v0_139) subv0 ON subt1.c_1_1 = subv0.c_1_0 AND subt1.c_1_1 < subv0.c_1_0,  v11 AS v11_142 INNER JOIN (SELECT v13_144.c_1_0 FROM v13 AS v13_144) subv13 ON v11_142.c_1_0 = subv13.c_1_0 AND '1969-12-13' <= subv13.c_1_0 AND v11_142.c_1_0 = subv13.c_1_0 AND v11_142.c_1_0 > subv13.c_1_0,  t0 AS t0_137 RIGHT ANTI JOIN v4 AS v4_148 ON t0_137.c_0_3 = v4_148.c_1_0 AND t0_137.c_0_3 != v4_148.c_1_0 AND t0_137.c_0_6 > v4_148.c_1_0 AND t0_137.c_0_6 < v4_148.c_1_0 WHERE CAST(subt1.c_1_1 AS BOOLEAN ) ;
CREATE VIEW v17 AS SELECT subt1.c_1_0 FROM v8 AS v8_152, v12 AS v12_143,  (SELECT t1_138.c_1_0, t1_138.c_1_1 FROM t1 AS t1_138 WHERE ('1969-12-13') IN ((SELECT v14_145.c_1_0 FROM v14 AS v14_145, v5 AS v5_149, v3 AS v3_147, v1 AS v1_140 WHERE NULL) ) ) subt1 INNER JOIN (SELECT v0_139.c_1_0 FROM v0 AS v0_139) subv0 ON subt1.c_1_1 = subv0.c_1_0 AND subt1.c_1_1 < subv0.c_1_0,  v11 AS v11_142 INNER JOIN (SELECT v13_144.c_1_0 FROM v13 AS v13_144) subv13 ON v11_142.c_1_0 = subv13.c_1_0 AND '1969-12-13' <= subv13.c_1_0 AND v11_142.c_1_0 = subv13.c_1_0 AND v11_142.c_1_0 > subv13.c_1_0,  t0 AS t0_137 RIGHT ANTI JOIN v4 AS v4_148 ON t0_137.c_0_3 = v4_148.c_1_0 AND t0_137.c_0_3 != v4_148.c_1_0 AND t0_137.c_0_6 > v4_148.c_1_0 AND t0_137.c_0_6 < v4_148.c_1_0 WHERE (NOT (CAST(subt1.c_1_1 AS BOOLEAN ) ));
CREATE VIEW v18 AS SELECT subt1.c_1_0 FROM v8 AS v8_152, v12 AS v12_143,  (SELECT t1_138.c_1_0, t1_138.c_1_1 FROM t1 AS t1_138 WHERE ('1969-12-13') IN ((SELECT v14_145.c_1_0 FROM v14 AS v14_145, v5 AS v5_149, v3 AS v3_147, v1 AS v1_140 WHERE NULL) ) ) subt1 INNER JOIN (SELECT v0_139.c_1_0 FROM v0 AS v0_139) subv0 ON subt1.c_1_1 = subv0.c_1_0 AND subt1.c_1_1 < subv0.c_1_0,  v11 AS v11_142 INNER JOIN (SELECT v13_144.c_1_0 FROM v13 AS v13_144) subv13 ON v11_142.c_1_0 = subv13.c_1_0 AND '1969-12-13' <= subv13.c_1_0 AND v11_142.c_1_0 = subv13.c_1_0 AND v11_142.c_1_0 > subv13.c_1_0,  t0 AS t0_137 RIGHT ANTI JOIN v4 AS v4_148 ON t0_137.c_0_3 = v4_148.c_1_0 AND t0_137.c_0_3 != v4_148.c_1_0 AND t0_137.c_0_6 > v4_148.c_1_0 AND t0_137.c_0_6 < v4_148.c_1_0 WHERE ((CAST(subt1.c_1_1 AS BOOLEAN ) ) IS NULL);
CREATE VIEW v19 AS ((SELECT subt1.c_1_0 FROM v8 AS v8_152, v12 AS v12_143,  (SELECT t1_138.c_1_0, t1_138.c_1_1 FROM t1 AS t1_138 WHERE ('1969-12-13') IN ((SELECT v14_145.c_1_0 FROM v14 AS v14_145, v5 AS v5_149, v3 AS v3_147, v1 AS v1_140 WHERE NULL) ) ) subt1 INNER JOIN (SELECT v0_139.c_1_0 FROM v0 AS v0_139) subv0 ON subt1.c_1_1 = subv0.c_1_0 AND subt1.c_1_1 < subv0.c_1_0,  v11 AS v11_142 INNER JOIN (SELECT v13_144.c_1_0 FROM v13 AS v13_144) subv13 ON v11_142.c_1_0 = subv13.c_1_0 AND '1969-12-13' <= subv13.c_1_0 AND v11_142.c_1_0 = subv13.c_1_0 AND v11_142.c_1_0 > subv13.c_1_0,  t0 AS t0_137 RIGHT ANTI JOIN v4 AS v4_148 ON t0_137.c_0_3 = v4_148.c_1_0 AND t0_137.c_0_3 != v4_148.c_1_0 AND t0_137.c_0_6 > v4_148.c_1_0 AND t0_137.c_0_6 < v4_148.c_1_0 WHERE CAST(subt1.c_1_1 AS BOOLEAN ) ) UNION ALL (SELECT subt1.c_1_0 FROM v8 AS v8_152, v12 AS v12_143,  (SELECT t1_138.c_1_0, t1_138.c_1_1 FROM t1 AS t1_138 WHERE ('1969-12-13') IN ((SELECT v14_145.c_1_0 FROM v14 AS v14_145, v5 AS v5_149, v3 AS v3_147, v1 AS v1_140 WHERE NULL) ) ) subt1 INNER JOIN (SELECT v0_139.c_1_0 FROM v0 AS v0_139) subv0 ON subt1.c_1_1 = subv0.c_1_0 AND subt1.c_1_1 < subv0.c_1_0,  v11 AS v11_142 INNER JOIN (SELECT v13_144.c_1_0 FROM v13 AS v13_144) subv13 ON v11_142.c_1_0 = subv13.c_1_0 AND '1969-12-13' <= subv13.c_1_0 AND v11_142.c_1_0 = subv13.c_1_0 AND v11_142.c_1_0 > subv13.c_1_0,  t0 AS t0_137 RIGHT ANTI JOIN v4 AS v4_148 ON t0_137.c_0_3 = v4_148.c_1_0 AND t0_137.c_0_3 != v4_148.c_1_0 AND t0_137.c_0_6 > v4_148.c_1_0 AND t0_137.c_0_6 < v4_148.c_1_0 WHERE (NOT (CAST(subt1.c_1_1 AS BOOLEAN ) ))) UNION ALL (SELECT subt1.c_1_0 FROM v8 AS v8_152, v12 AS v12_143,  (SELECT t1_138.c_1_0, t1_138.c_1_1 FROM t1 AS t1_138 WHERE ('1969-12-13') IN ((SELECT v14_145.c_1_0 FROM v14 AS v14_145, v5 AS v5_149, v3 AS v3_147, v1 AS v1_140 WHERE NULL) ) ) subt1 INNER JOIN (SELECT v0_139.c_1_0 FROM v0 AS v0_139) subv0 ON subt1.c_1_1 = subv0.c_1_0 AND subt1.c_1_1 < subv0.c_1_0,  v11 AS v11_142 INNER JOIN (SELECT v13_144.c_1_0 FROM v13 AS v13_144) subv13 ON v11_142.c_1_0 = subv13.c_1_0 AND '1969-12-13' <= subv13.c_1_0 AND v11_142.c_1_0 = subv13.c_1_0 AND v11_142.c_1_0 > subv13.c_1_0,  t0 AS t0_137 RIGHT ANTI JOIN v4 AS v4_148 ON t0_137.c_0_3 = v4_148.c_1_0 AND t0_137.c_0_3 != v4_148.c_1_0 AND t0_137.c_0_6 > v4_148.c_1_0 AND t0_137.c_0_6 < v4_148.c_1_0 WHERE ((CAST(subt1.c_1_1 AS BOOLEAN ) ) IS NULL)) );
CREATE VIEW v20 AS SELECT t1_233.c_1_0 FROM t1 AS t1_233, v10 AS v10_236, v14 AS v14_240, v7 AS v7_251, v13 AS v13_239, t0 AS t0_232, v4 AS v4_248, v12 AS v12_238, v3 AS v3_247, v8 AS v8_252, v18 AS v18_244, v17 AS v17_243, v0 AS v0_234;
CREATE VIEW v21 AS SELECT t1_233.c_1_0 FROM t1 AS t1_233, v10 AS v10_236, v14 AS v14_240, v7 AS v7_251, v13 AS v13_239, t0 AS t0_232, v4 AS v4_248, v12 AS v12_238, v3 AS v3_247, v8 AS v8_252, v18 AS v18_244, v17 AS v17_243, v0 AS v0_234 WHERE CAST([-1991726913, 1312178014] AS BOOLEAN ) ;
CREATE VIEW v21 AS SELECT v20_292.c_1_0 FROM v20 AS v20_292, v8 AS v8_298, v1 AS v1_280, v0 AS v0_279;
CREATE VIEW v65 AS SELECT v20_292.c_1_0 FROM v20 AS v20_292, v8 AS v8_298, v1 AS v1_280, v0 AS v0_279 WHERE false;
CREATE VIEW v22 AS SELECT v20_292.c_1_0 FROM v20 AS v20_292, v8 AS v8_298, v1 AS v1_280, v0 AS v0_279 WHERE (NOT (false));
CREATE VIEW v23 AS SELECT v20_292.c_1_0 FROM v20 AS v20_292, v8 AS v8_298, v1 AS v1_280, v0 AS v0_279 WHERE ((false) IS NULL);
CREATE VIEW v24 AS ((SELECT v20_292.c_1_0 FROM v20 AS v20_292, v8 AS v8_298, v1 AS v1_280, v0 AS v0_279 WHERE false) UNION ALL (SELECT v20_292.c_1_0 FROM v20 AS v20_292, v8 AS v8_298, v1 AS v1_280, v0 AS v0_279 WHERE (NOT (false))) UNION ALL (SELECT v20_292.c_1_0 FROM v20 AS v20_292, v8 AS v8_298, v1 AS v1_280, v0 AS v0_279 WHERE ((false) IS NULL)) );
CREATE VIEW v25 AS SELECT t0_402.c_0_0 FROM t0 AS t0_402, v24 AS v24_421, v0 AS v0_404, v13 AS v13_409, v12 AS v12_408, v11 AS v11_407, v3 AS v3_422, v9 AS v9_429, v5 AS v5_424, v22 AS v22_419, t1 AS t1_403, v6 AS v6_425, v2 AS v2_416, v19 AS v19_415;
CREATE VIEW v26 AS SELECT t0_402.c_0_0 FROM t0 AS t0_402, v24 AS v24_421, v0 AS v0_404, v13 AS v13_409, v12 AS v12_408, v11 AS v11_407, v3 AS v3_422, v9 AS v9_429, v5 AS v5_424, v22 AS v22_419, t1 AS t1_403, v6 AS v6_425, v2 AS v2_416, v19 AS v19_415 WHERE false;
CREATE VIEW v27 AS SELECT t0_402.c_0_0 FROM t0 AS t0_402, v24 AS v24_421, v0 AS v0_404, v13 AS v13_409, v12 AS v12_408, v11 AS v11_407, v3 AS v3_422, v9 AS v9_429, v5 AS v5_424, v22 AS v22_419, t1 AS t1_403, v6 AS v6_425, v2 AS v2_416, v19 AS v19_415 WHERE (NOT (false));
CREATE VIEW v28 AS SELECT t0_402.c_0_0 FROM t0 AS t0_402, v24 AS v24_421, v0 AS v0_404, v13 AS v13_409, v12 AS v12_408, v11 AS v11_407, v3 AS v3_422, v9 AS v9_429, v5 AS v5_424, v22 AS v22_419, t1 AS t1_403, v6 AS v6_425, v2 AS v2_416, v19 AS v19_415 WHERE ((false) IS NULL);
CREATE VIEW v29 AS ((SELECT t0_402.c_0_0 FROM t0 AS t0_402, v24 AS v24_421, v0 AS v0_404, v13 AS v13_409, v12 AS v12_408, v11 AS v11_407, v3 AS v3_422, v9 AS v9_429, v5 AS v5_424, v22 AS v22_419, t1 AS t1_403, v6 AS v6_425, v2 AS v2_416, v19 AS v19_415 WHERE false) UNION ALL (SELECT t0_402.c_0_0 FROM t0 AS t0_402, v24 AS v24_421, v0 AS v0_404, v13 AS v13_409, v12 AS v12_408, v11 AS v11_407, v3 AS v3_422, v9 AS v9_429, v5 AS v5_424, v22 AS v22_419, t1 AS t1_403, v6 AS v6_425, v2 AS v2_416, v19 AS v19_415 WHERE (NOT (false))) UNION ALL (SELECT t0_402.c_0_0 FROM t0 AS t0_402, v24 AS v24_421, v0 AS v0_404, v13 AS v13_409, v12 AS v12_408, v11 AS v11_407, v3 AS v3_422, v9 AS v9_429, v5 AS v5_424, v22 AS v22_419, t1 AS t1_403, v6 AS v6_425, v2 AS v2_416, v19 AS v19_415 WHERE ((false) IS NULL)) );
CREATE VIEW v30 AS SELECT v2_566.c_1_0 FROM v2 AS v2_566, v65 AS v65_581, t0 AS t0_552, v20 AS v20_567, v4 AS v4_578, v0 AS v0_554, v6 AS v6_580, v29 AS v29_576, v3 AS v3_577, v19 AS v19_565, v23 AS v23_570, v8 AS v8_583, v22 AS v22_569, v7 AS v7_582, v16 AS v16_562, v18 AS v18_564, v5 AS v5_579, t1 AS t1_553, v12 AS v12_558, v17 AS v17_563, v21 AS v21_568, v28 AS v28_575, v1 AS v1_555, v11 AS v11_557, v24 AS v24_571, v25 AS v25_572, v9 AS v9_584, v26 AS v26_573, v15 AS v15_561, v14 AS v14_560, v13 AS v13_559, v10 AS v10_556;

SELECT * FROM v30;
```

### Expected behavior (Required)
no error

### Real behavior (Required)

```Prolog
2022-12-13 23:29:06,754 WARN (starrocks-mysql-nio-pool-3|225) [StmtExecutor.execute():519] execute Exception, sql SELECT * FROM v30;
java.lang.IllegalStateException: null
        at com.google.common.base.Preconditions.checkState(Preconditions.java:494) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.qe.Coordinator$FragmentExecParams.toThriftForUniqueParams(Coordinator.java:3199) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator$FragmentExecParams.toThriftInBatch(Coordinator.java:3322) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator.deliverExecBatchFragmentsRequests(Coordinator.java:1126) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator.deliverExecFragments(Coordinator.java:750) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.Coordinator.exec(Coordinator.java:654) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.handleQueryStmt(StmtExecutor.java:726) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:428) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:312) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:427) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:673) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 +-------------------------+
| current_version()       |
+-------------------------+
| BRANCH-2.5-ASAN 3609c26 |
+-------------------------+

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15183.

## Problem Summary(Required) ：

`MultiCastPlanFragment` appends destinations to `MultiCastPlanFragment.dataSink.destinations` in `Coordinator::prepareFragments`.

However, when execution of this query fails and retries, destinations will be appended multiple times.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|1|5|20.00%|[310, 311, 312, 313]|

File: fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java [hunk 1]
```diff
@@ -304,6 +304,15 @@ public void prepareExec() throws Exception {
 
     private void prepareFragments() {
         for (PlanFragment fragment : fragments) {
+            // Clear each destination for multi sink to avoid add destinations multiple times
+            // when retrying the query with these fragments.
+            if (fragment instanceof MultiCastPlanFragment) {
```
[the remark this answers is not in the document]
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java:311]: Why not reset other Fragment type?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 9 (55.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|5|9|55.56%|[312, 313, 314, 315]|

File: fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java:311]: @LiShuMing  Only `MultiCastPlanFragment` has `MultiDataSink`, which stores state in the fragment.

[the remark this answers is not in the document]
Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPreprocessor.java:311]: @kangkaisen Fixed.

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**